Packs/RSANetWitness_v11_1/Integrations/RSANetWitnessv115/RSANetWitnessv115_test.py

```diff
@@ -347,3 +350,182 @@ def test_paging_command(mocker):
+
+
+@pytest.mark.parametrize(
+    'xsoar_status, xsoar_close_reason, expected_rsa_status',
+    [
+        (1, None, "New"),
+        (2, None, "Closed"),
+        (2, "False positive", "ClosedFalsePositive"),
+    ]
+)
+def test_xsoar_status_to_rsa_status(mocker, xsoar_status: int, xsoar_close_reason: str | None, expected_rsa_status: str):
+    """
+    Given:
+        -  XSOAR Status int
+        -  XSOAR Close reason
+    When
+        - running xsoar_status_to_rsa_status
+    Then
+        - Return RSA Status
+    """
+    assert xsoar_status_to_rsa_status(xsoar_status, xsoar_close_reason) == expected_rsa_status
+
+
+def test_update_remote_system_command(mocker):
+    """
+        Given:
+        - client with fetch parameters
+        - args with incident ID, status, delta_keys
+        - param no used by integration
+
+        When:
+            Calling the update_remote_system_command.
+
+        Then:
+            Check if RSA status has been updated.
+    """
+    paging_data = "INC-1"
+
+    class UpdateRemoteSystemArgsResponse:
+        def __init__(self) -> dict:
+            self.delta = {"key": "value"}
+            self.remote_incident_id = "INC-1"
+            self.data = {"status": 1, "closeReason": None}
+
+    mocker.patch.object(RSANetWitnessv115, "UpdateRemoteSystemArgs", return_value=UpdateRemoteSystemArgsResponse())
+    mocker.patch.object(client, "get_incident_request", return_value={"id": "INC-1", "status": 1})
+    mocker.patch.object(client, "update_incident_request", return_value={"id": "INC-1", "status": 1, "assignee": None})
+
+    assert update_remote_system_command(client, {}, {}) == paging_data
+
+
+def test_get_remote_data_command(mocker):
+    """
+        Given:
+        - client with fetch parameters
+        - args with incident attributes
+        - params
+
+        When:
+            running get_remote_data_command.
+
+        Then:
+            Update context of incident if incident has been updated from RSA.
+    """
+    paging_data = {'alertCount': 2, 'alerts': {'alerts': [{'INC-2'}]}, 'id': 1, 'status': 1}
+
+    class GetRemoteDataArgsResponse:
+        def __init__(self) -> dict:
+            self.last_update = 1234567890
+            self.remote_incident_id = 0
+
+    mocker.patch.object(RSANetWitnessv115, "GetRemoteDataArgs", return_value=GetRemoteDataArgsResponse())
+    mocker.patch.object(RSANetWitnessv115, "argToBoolean", return_value=True)
+    mocker.patch.object(RSANetWitnessv115, "arg_to_number", return_value=2)
+    mocker.patch.object(client, "get_incident_request", return_value={"id": 1, "status": 1, "alertCount": 2})
+    mocker.patch.object(RSANetWitnessv115, "fetch_alerts_related_incident", return_value={"alerts": [{"INC-2"}]})
+
+    res = get_remote_data_command(client, {}, {'close_incident': 0, 'import_alerts': True, 'max_alerts': 1})
+    assert res.mirrored_object == paging_data
+
+
+def test_get_modified_remote_data_command(mocker):
```

Comment:
Please add a use case in which the incident is in the integration context and needs to be updated in xsoar.

Reply:
I created others functions to add some tests:
- test_get_remote_data_command_with_closed_xsoar_incident
- test_get_remote_data_command_with_new_xsoar_incident_entries

Reply:
What do you mean by "in the integration context", when the "integration context" is already in xsoar.

The aim here is to check if the RSA return a incident who need to be updated in xsoar, then return the RSA Incident ID.